### PR TITLE
feat: Join a conversation with a guest link [AR-2703]

### PR DESCRIPTION
**Pull request by MohamadJaara**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/AR-2703" title="AR-2703" target="_blank">StoryAR-2703</a>  Join a conversation with a guest link - Wire installed + account exists
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
    - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
    - [ ] contains a reference JIRA issue number like `SQPIT-764`
    - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
    - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

[AR-2703]

### Solutions

handle join conversation deep link
on join conversation intent received 
1. check if the code is valid and if self is already a member
2. if code is valid and self is not a member then display a dialog for user confirm
3. on confirm join the conversion and navigate to the conversation screen 
4. if self already a member then navigate to the conversation screen

### Dependencies (Optional)

_If there are some other pull requests related to this one (e.g. new releases of frameworks), specify them here._

Needs releases with:

- [ ] GitHub link to other pull request

#### Test Coverage (Optional)

- [ ] I have added automated test to this contribution

#### How to Test

_Briefly describe how this change was tested and if applicable the exact steps taken to verify that it works as expected._

### Notes (Optional)

_Specify here any other facts that you think are important for this issue._

### Attachments (Optional)

_Attachments like images, videos, etc. (drag and drop in the text box)_ 

----
#### PR Post Submission Checklist for internal contributors (Optional)

- [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

- [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

[AR-2703]: https://wearezeta.atlassian.net/browse/AR-2703?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [1](https://10.10.124.134/job/wire-android-reloaded/job/PR-1294/1/) **failed**.

**Comment by AndroidBob** [comment]
Build [3](https://10.10.124.134/job/wire-android-reloaded/job/PR-1294/3/) **failed**.

**Comment by AndroidBob** [comment]
Build [4](https://10.10.124.134/job/wire-android-reloaded/job/PR-1294/4/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(5442938)(5442938)-dev-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/compat/com.wire.android-vv0.0.1(5442938)(5442938)-dev-compat.apk)

**Comment by AndroidBob** [comment]
Build [5](https://10.10.124.134/job/wire-android-reloaded/job/PR-1294/5/) **failed**.

**Comment by AndroidBob** [comment]
Build [6](https://10.10.124.134/job/wire-android-reloaded/job/PR-1294/6/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(5610944)(5610944)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv0.0.1(5610944)(5610944)-staging-compat.apk)

**Comment by AndroidBob** [comment]
Build [7](https://10.10.124.134/job/wire-android-reloaded/job/PR-1294/7/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.0.0(5620882)(5620882)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv4.0.0(5620882)(5620882)-staging-compat.apk)

**Comment by AndroidBob** [comment]
Build [8](https://10.10.124.134/job/wire-android-reloaded/job/PR-1294/8/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.0.0(5620997)(5620997)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv4.0.0(5620997)(5620997)-staging-compat.apk)

**Comment by AndroidBob** [comment]
Build [9](https://10.10.124.134/job/wire-android-reloaded/job/PR-1294/9/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.0.0(5628469)(5628469)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv4.0.0(5628469)(5628469)-staging-compat.apk)

**Comment by AndroidBob** [comment]
Build [10](https://10.10.124.134/job/wire-android-reloaded/job/PR-1294/10/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.0.0(5628890)(5628890)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv4.0.0(5628890)(5628890)-staging-compat.apk)

File: app/src/main/kotlin/com/wire/android/ui/WireActivityViewModel.kt [hunk 1]
```diff
@@ -386,6 +376,80 @@ class WireActivityViewModel @Inject constructor(
...
+                        }
+
+                        is CheckConversationInviteCodeUseCase.Result.Failure -> globalAppState =
+                            globalAppState.copy(conversationJoinedDialog = JoinConversationViaCodeState.Error(result))
+                    }
+                }
+            }
+        }
+
+    fun joinConversationViaCode(
```
Comment (ohassine) [inline app/src/main/kotlin/com/wire/android/ui/WireActivityViewModel.kt:416]: Can we also cover error case with test ? when `currentSession` and `joinConversationViaCode` return Failure

**Comment by AndroidBob** [comment]
Build [11](https://10.10.124.134/job/wire-android-reloaded/job/PR-1294/11/) **failed**.

**Comment by AndroidBob** [comment]
Build [14](https://10.10.124.134/job/wire-android-reloaded/job/PR-1294/14/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.0.0(5673139)(5673139)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv4.0.0(5673139)(5673139)-staging-compat.apk)

**Comment by AndroidBob** [comment]
Build [15](https://10.10.124.134/job/wire-android-reloaded/job/PR-1294/15/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.0.0(5678803)(5678803)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv4.0.0(5678803)(5678803)-staging-compat.apk)

**Review by ohassine — APPROVED** [review]
lgtm 👌

**Pull request merged**